src/main/java/picard/illumina/parser/readers/CbclReader.java

```diff
@@ -332,28 +332,31 @@ private void cacheTile(final int totalCycleCount, final TileData tileData, final
                     (totalCycleCount + 1), this.streamFiles[totalCycleCount].getAbsolutePath()));
         }
 
-        // Uncompress the data from the buffer we just wrote - use gzip input stream to write to uncompressed buffer
+        // Decompress the data from the buffer we just wrote - use gzip input stream to write to uncompressed buffer
         final ByteArrayInputStream byteInputStream = new ByteArrayInputStream(Arrays.copyOfRange(tileByteArray, 0, readBytes));
 
-        final GZIPInputStream gzipInputStream = new GZIPInputStream(byteInputStream, uncompressedByteArray.length);
-        int read;
-        int totalRead = 0;
-        try {
-            while ((read = gzipInputStream.read(uncompressedByteArray, totalRead, uncompressedByteArray.length - totalRead)) != -1) {
-                if (read == 0) break;
-                totalRead += read;
+        //only decompress the data if we are expecting data.
+        if (uncompressedByteArray.length > 0) {
+            int read;
+            int totalRead = 0;
+            try (GZIPInputStream gzipInputStream = new GZIPInputStream(byteInputStream, uncompressedByteArray.length)) {
+                while ((read = gzipInputStream.read(uncompressedByteArray, totalRead, uncompressedByteArray.length - totalRead)) != -1) {
+                    if (read == 0) break;
+                    totalRead += read;
+                }
+            } catch (final EOFException eofException) {
+                throw new PicardException("Unexpected end of file " + this.streamFiles[totalCycleCount].getAbsolutePath()
+                        + " this file is likely corrupt or truncated. We have read "
+                        + totalRead + " and were expecting to read "
+                        + uncompressedByteArray.length);
             }
-        } catch (final EOFException eofException) {
-            throw new PicardException("Unexpected end of file " + this.streamFiles[totalCycleCount].getAbsolutePath()
-                    + " this file is likely corrupt or truncated. We have read "
-                    + totalRead + " and were expecting to read "
-                    + uncompressedByteArray.length);
-        }
-        if (totalRead != tileData.uncompressedBlockSize) {
-            throw new PicardException(String.format("Error while decompressing from BCL file for cycle %d. Offending file on disk is %s",
-                    (totalCycleCount + 1), this.streamFiles[totalCycleCount].getAbsolutePath()));
+            if (totalRead != tileData.uncompressedBlockSize) {
+                throw new PicardException(String.format("Error while decompressing from BCL file for cycle %d. Offending file on disk is %s",
+                        (totalCycleCount + 1), this.streamFiles[totalCycleCount].getAbsolutePath()));
+            }
+        } else {
+            log.warn("Ignoring tile " + tileData.tileNum + " there are no PF reads.");
```

Comment:
If this was in a method, you could do the negative case first, with a `return`, which IMO would be clearer.